src/pages/chart.jsx

```diff
@@ -0,0 +1,22 @@
+import EmbedPlayer from "../components/embedPlayer";
+
+const domain = process.env.NEXT_PUBLIC_EMBED_DOMAIN_URL;
+
+export async function getServerSideProps(context) {
+  const { query } = context;
+  const queryString = Object.keys(query)
+    .map((key) => `${key}=${query[key]}`)
+    .join("&");
+
+  const result = await fetch(`${domain}/api/chart?${queryString}`);
+
+  const data = await result.json();
+
+  return { props: { trackData: data } };
```

Comment:
should we check that the response is successful and render an error page if it isn't?

Reply:
could use a redirect if we dont get a success response
```suggestion
  const data = await result.json();.
  
  if (!data) {
    return {
      redirect: {
        destination: '/',
        permanent: false,
      },
    }
  }
  
  return { props: { trackData: data } };
```